### DISCUSSION TITLE
Not able to see the last pushed version on github

### DISCUSSION THREAD

**Discussion by najeeb1010** [community/community#23268] [New to GitHub]
<p>I have been an ardent subversion user for years, and became a convert to git only just recently. Consequently, I’m still not very comfortable moving around in git, so please bear with me if I may sound like a noob.</p>
<p>I have a private repository on github. I recently did a whole load of work on my local machine and then realized that I may not want to push or commit them yet because they are not priority features.</p>
<p>I therefore checked out another copy of the repoistory on a different location on my system, keeping the old location on my local machine intact, and then working off the newly checked out repo. I made the required changes to this new location on my local machine, adding the priority features there, and then pushed those changes to github, and then checked the web interface and github’s reflecting those changes, all right.</p>
<p>Now I have gone back to the old location in my local machine to complete the non-priority features, and when I do a diff, I see that it’s still comparing it with the older version on github; that is, it’s not reflecting the last changes that I pushed there.</p>
<p>This is quite confusing to me, since I am used to seeing the most recent changes when I do a diff on subversion.</p>
<p>And yes, I’m using the TortoiseGit client on my local machine.</p>
<p>Any help will be appreciated. Thanks!</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Git is a decentralized system, which mean your local copy of the repository is complete in itself, and actions like “diff” that aren’t about interaction with other copies of the repository (like on GitHub or your second directory) won’t connect to any other repositories, and just work on the local data. Note also that by default (without any parameters setting what to compare) “git diff” will compare your “working copy” (files in the repository directory) with the latest commit on your current branch, regardless of that branch’s connection to any other branches or repositories.</p>
<p>If you want to get the latest version from your GitHub repository, you have to <em>fetch</em> the latest state first. If you created the repository by cloning from GitHub just calling “git fetch” will be enough, otherwise you’ll have to give the name of the “remote” (remote repository). A git repository can have many remotes, you could even set your second local repository as one and fetch from there.</p>
<p>The further procedure depends on what exactly you want, and whether you have locally committed any of your changes already. If you haven’t committed anything and are on a branch tracking the GitHub repository (“master” does by default after a clone), the easiest thing is likely going to be the following:</p>
<ol>
<li>“git stash”: This saves your changes in a “stash” (basically a diff on a stack, you can have multiple stashes that way), and removes them from the working copy.</li>
<li>“git pull”: A combined fetch and merge from the upstream tracking branch. This should get your local files to the same state as the same branch in your GitHub repository.</li>
<li>“git stash pop”: The reverse of step one, apply the stashed changes to the working copy and remove it from the stack.</li>
</ol>
<p>If step 3 causes any conflicts with the changes you’ve pulled you’ll need to resolve them manually.</p>
